h5g_group.go

```diff
@@ -162,3 +163,15 @@ func (g *Group) CreateTableFrom(name string, dtype interface{}, chunkSize, compr
 func (g *Group) OpenTable(name string) (*Table, error) {
 	return openTable(g.id, name)
 }
+
+// CheckLink checks if a link( can be group or dataset or actual link )exsit or not. This method won't give you annoying hdf5 warnings when called in a goroutine
+// CheckLink returns nil when a link exist, return an error when the link doesn't exist or some other error occured
```

Comment:
```
// LinkExists returns whether a link with the specified name exists in the group.
func (g *CommonFG) LinkExists(name string) bool {
	c_name := C.CString(name)
	defer C.free(unsafe.Pointer(c_name))
	return C.H5Lexists(g.id, c_name, 0) > 0
}
```

The HDF5 API is discusting for this. The return status, according to [this](https://support.hdfgroup.org/HDF5/doc/RM/RM_H5L.html#Link-Exists), is almost entirely uninformative for the failing case, so I'm not sure that it's even worth bothering to indicate the error return. @sbinet?

Reply:
The reason why I chose an error instead of  bool is that in the h5group said" "Returns a negative value when the function fails and may return a negative value if the link does not exist". So looks like there are 3 cases: exist, doesn't exist, and some error. A bool is totally OK, exist or other than exist.

Reply:
>So looks like there are 3 cases: exist, doesn't exist, and some error. A bool is totally OK, exist or other than exist.

Yes, but in the linked docs, we see this 'Returns a negative value when the function fails and *may return a negative value if the link does not exist*. See “Failure Modes” below.', so negative may mean zero. Given that the API docs don't specify what the different negative values actually mean, it's not obvious how that information would be used (and in the current state of this PR, you already throw away the difference between these).

Reply:
OK. That make sense. Do you like me to make the change to bool and push to this PR?

Reply:
Yes, please.

Reply:
yes, a `bool` might better reflect what's happening behind the curtains of the underlying C library.